packages/ppx/src/declarations_to_emotion.re

```diff
@@ -950,7 +952,129 @@ let render_color = (~loc, value) =>
   | `Function_hsl(_)
   | `Function_hsla(_)
   | `Deprecated_system_color(_) => raise(Unsupported_feature)
+  }
+
+and render_function_color_mix = (~loc, value: Types.function_color_mix) => {
+  let render_rectangular_color_space = (x: Types.rectangular_color_space) => {
+    switch (x) {
+    | `Srgb => [%expr `srgb]
+    | `Srgb_linear => [%expr `srgbLinear]
+    | `Display_p3 => [%expr `displayP3]
+    | `A98_rgb => [%expr `a98Rgb]
+    | `Prophoto_rgb => [%expr `prophotoRgb]
+    | `Rec2020 => [%expr `rec2020]
+    | `Lab => [%expr `lab]
+    | `Oklab => [%expr `oklab]
+    | `Xyz => [%expr `xyz]
+    | `Xyz_d50 => [%expr `xyzD50]
+    | `Xyz_d65 => [%expr `xyzD65]
+    };
+  };
+  let render_polar_color_space = (x: Types.polar_color_space) => {
+    switch (x) {
+    | `Hsl => [%expr `hsl]
+    | `Hwb => [%expr `hwb]
+    | `Lch => [%expr `lch]
+    | `Oklch => [%expr `oklch]
+    };
+  };
+
+  let render_size_hue = (x: Types.hue_interpolation_method) => {
+    let render_size =
+      fun
+      | `Shorter => [%expr `shorter]
+      | `Longer => [%expr `longer]
+      | `Increasing => [%expr `increasing]
+      | `Decreasing => [%expr `decreasing];
+
+    switch (x) {
+    // Should it be a unit 🤔?
+    | (x, ()) => render_size(x)
+    };
+  };
+
+  let render_in = [%expr `in_];
+
+  switch (value) {
+  // Should it be a unit 🤔?
+  | (x, (), colors: list((Types.color, option(Types.percentage)))) =>
+    let ((color_one, percentage_one), (color_two, percentage_two)) =
+      switch (colors) {
+      | [(c1, p1), (c2, p2)] => ((c1, p1), (c2, p2))
+      | _ => failwith("unreachable")
+      };
+
+    /*
+     https://drafts.csswg.org/css-color-5/#color-mix-percent-norm
+
+      - If p1 + p2 ≠ 100%, then p1' = p1 / (p1 + p2) and p2' = p2 / (p1 + p2),
+          where p1' and p2' are the normalization results.
+      - If p1 = p2 = 0%, the function is invalid
+      - If both percentage, p1 and p2 are ommited, then p1 = p2 = 50%.
+      - If p1 is omitted, then p1 = 100% - p2.
+      - If p2 is omitted, then p2 = 100% - p1 */
+
+    let render_percentage = (p1, p2) => {
```

Comment:
let render_percentage = (p1, p2) => {
  switch (p1, p2) {
  | (Some(p1'), Some(p2')) when p1 == 0. and p2' == 0. => raise(Unsupported_feature)
  | (Some(p1'), Some(p2')) when p1' + p2' == 100. => render_percentage(~loc, p1' /. (p1' +. p2'))
  | (Some(p1'), Some(p2')) => render_percentage(~loc, p1')
  | (Some(p1'), None) => render_percentage(~loc, p1')
  | (None, Some(p2')) => render_percentage(~loc, 100. -. p2')
  | (None, None) => render_percentage(~loc, 50.)
  };
};